x-pack/plugin/text-structure/src/main/java/org/elasticsearch/xpack/textstructure/transport/TransportFindFieldStructureAction.java

```diff
@@ -47,18 +45,9 @@ public TransportFindFieldStructureAction(
         TransportService transportService,
         ActionFilters actionFilters,
         Client client,
-        ThreadPool threadPool,
-        TextStructExecutor executor
+        ThreadPool threadPool
     ) {
-        super(
-            FindFieldStructureAction.NAME,
-            transportService,
-            actionFilters,
-            FindFieldStructureAction.Request::new,
-            // workaround for https://github.com/elastic/elasticsearch/issues/97916
-            // TODO when the above issue is fixed, change this back to threadPool.generic()
-            DIRECT_EXECUTOR_SERVICE
-        );
+        super(FindFieldStructureAction.NAME, transportService, actionFilters, FindFieldStructureAction.Request::new, threadPool.generic());
```

Comment:
This is the only "real" change, right? (And the remainder is just some minor refactoring.)